### PR TITLE
Feature/test write box

### PR DESCRIPTION
**Pull request by ian-spoonradio**
- Add test code for all leaves in box tree
- Remove entry_count that is duplicated with `entries.len()`

### CODE REVIEW

**Comment by alfg** [comment]
Thank you! 👍

File: Cargo.toml [hunk 1]
```diff
@@ -20,3 +20,4 @@ license = "MIT"
 [dependencies]
 thiserror = "^1.0"
 byteorder = "1"
+num-rational = "0.3"
```
Comment (alfg) [inline Cargo.toml:23]: Just missed this. But this means `Cargo.lock` needs an update too, right?

Comment (ian-spoonradio) [reply Cargo.toml:23]: Yes, I missed it.
But, I am not sure if it is OK to keep `Cargo.lock`.
https://doc.rust-lang.org/cargo/faq.html#why-do-binaries-have-cargolock-in-version-control-but-not-libraries

Comment (alfg) [reply Cargo.toml:23]: Ah, I wasn't aware it wasn't recommended for libraries. I've removed the Cargo.lock and added it to .gitignore. Thanks for the clarification.

**Pull request merged**